### PR TITLE
x11-terms/ptyxis: New Package 46.5, x11-libs/vte, gui-libs/vte, gui-libs/vte-common: Bump to 0.76.3, x11-terms/gnome-terminal: Bump to 3.52.2

### PR DESCRIPTION
**Pull request by flint2**

### CODE REVIEW

**Comment by leio** [comment]
This should really depend on a vte-0.75.x version as well to have any kind of reasonable performance at runtime as per discussions with upstream

**Comment by flint2** [comment]
> This should really depend on a vte-0.75.x version as well to have any kind of reasonable performance at runtime as per discussions with upstream

Added VTE 0.75.0 ebuild with new prompt IUSE 

Compiling and working, performance needs testing

Still working on !vanilla patches rebase

**Comment by pacho2** [comment]
> Still working on !vanilla patches rebase

Hello,

is this still pending? (it seems it was rebased, but maybe I am looking to the wrong patch and that is the reason this PR is still tagged as "draft")

Thanks

File: gui-libs/vte/vte-0.76.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+	use elibc_musl && eapply "${FILESDIR}"/${PN}-0.66.2-musl-W_EXITCODE.patch
+
+	if ! use vanilla; then
+		# From https://src.fedoraproject.org/rpms/vte291/raw/rawhide/f/vte291-cntnr-precmd-preexec-scroll.patch
+		# Adds OSC 777 support for desktop notifications in gnome-terminal or elsewhere
+		eapply "${WORKDIR}"/${PN}-0.74.0-command-notify.patch
+	fi
+
+	if use ptyxis; then
```
Comment (pacho2) [inline gui-libs/vte/vte-0.76.3.ebuild:64]: Why don't apply them when USE vanilla is disabled? I would also add a note explaining that they come from Fedora (to also control better the updates/fixes that they could have). Maybe we can simply list the same references as they do:
https://src.fedoraproject.org/rpms/vte291/blob/f40/f/vte291.spec#_24

Also, aren't those patches replacing the vte-0.74.0-command-notify.patch one?

File: x11-terms/ptyxis/ptyxis-46.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+EAPI=8
+
+inherit gnome.org gnome2-utils meson xdg
+
+DESCRIPTION="Ptyxis is a terminal for GNOME with first-class support for containers."
+HOMEPAGE="https://gitlab.gnome.org/chergert/ptyxis"
+
+LICENSE="LGPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~loong"
```
Comment (pacho2) [inline x11-terms/ptyxis/ptyxis-46.3.ebuild:13]: If you only tested it on amd64, please drop the other KEYWORDS

File: x11-terms/ptyxis/ptyxis-46.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	>=dev-libs/glib-2.76:2
+	>=gui-libs/gtk-4.12.2:4
+	>=gui-libs/libadwaita-1.4_alpha:1
+	>=gui-libs/vte-0.75.0:2.91-gtk4[ptyxis]
```
Comment (pacho2) [inline x11-terms/ptyxis/ptyxis-46.3.ebuild:21]: Here it would need to be updated to !vanilla

File: x11-terms/ptyxis/ptyxis-46.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit gnome.org gnome2-utils meson xdg
+
+DESCRIPTION="Ptyxis is a terminal for GNOME with first-class support for containers."
```
Comment (pacho2) [inline x11-terms/ptyxis/ptyxis-46.3.ebuild:8]: Maybe a more direct description as the one provided in the HOMEPAGE would be a bit better:
"A terminal for a container-oriented desktop"

File: gui-libs/vte/vte-0.76.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,108 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{10..13} )
+
+inherit gnome.org meson python-any-r1 vala xdg
+
+DESCRIPTION="Library providing a virtual terminal emulator widget"
+HOMEPAGE="https://wiki.gnome.org/Apps/Terminal/VTE"
```
Comment (pacho2) [inline gui-libs/vte/vte-0.76.3.ebuild:10]: Maybe replace by:
https://gitlab.gnome.org/GNOME/vte

File: gui-libs/vte-common/vte-common-0.76.3.ebuild [hunk 6]
```diff
@@ -0,0 +1,81 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{10..13} )
+
+inherit gnome.org meson python-any-r1
+
+DESCRIPTION="Library providing a virtual terminal emulator widget"
+HOMEPAGE="https://wiki.gnome.org/Apps/Terminal/VTE"
```
Comment (pacho2) [inline gui-libs/vte-common/vte-common-0.76.3.ebuild:10]: https://gitlab.gnome.org/GNOME/vte

File: x11-libs/vte/vte-0.76.3.ebuild [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+"
+
+src_prepare() {
+	default
+	use vala && vala_setup
+	xdg_environment_reset
+
+	use elibc_musl && eapply "${FILESDIR}"/${PN}-0.66.2-musl-W_EXITCODE.patch
+
+	if ! use vanilla; then
```
Comment (pacho2) [inline x11-libs/vte/vte-0.76.3.ebuild:60]: It seems we could use the same patches as got the gtk4 version, but passing the configure options for gtk3:
https://src.fedoraproject.org/rpms/vte291/blob/f40/f/0001-a11y-implement-GtkAccessibleText.patch
https://src.fedoraproject.org/rpms/vte291/blob/f40/f/0001-add-notification-and-shell-precmd-preexec.patch

**Comment by flint2** [comment]
> > Still working on !vanilla patches rebase
> 
> Hello,
> 
> is this still pending? (it seems it was rebased, but maybe I am looking to the wrong patch and that is the reason this PR is still tagged as "draft")
> 
> Thanks

Added gnome-terminal 3.52.2 and patch to work with patched vte 
Consolidated vanilla use flag

Marked PR as ready for review

File: gui-libs/vte/vte-0.76.3.ebuild (hunk 1 above)
Comment (flint2) [reply gui-libs/vte/vte-0.76.3.ebuild]: Done in https://github.com/gentoo/gentoo/pull/34844/commits/4823224d61b12fd97a25efcf977885dc04d7904f

File: x11-terms/ptyxis/ptyxis-46.3.ebuild (hunk 2 above)
Comment (flint2) [reply x11-terms/ptyxis/ptyxis-46.3.ebuild]: Done in https://github.com/gentoo/gentoo/pull/34844/commits/50c97d53444bad308d41d3a12616af34fc17a1b3

File: x11-terms/ptyxis/ptyxis-46.3.ebuild (hunk 3 above)
Comment (flint2) [reply x11-terms/ptyxis/ptyxis-46.3.ebuild]: Done in https://github.com/gentoo/gentoo/pull/34844/commits/50c97d53444bad308d41d3a12616af34fc17a1b3

File: gui-libs/vte/vte-0.76.3.ebuild (hunk 5 above)
Comment (flint2) [reply gui-libs/vte/vte-0.76.3.ebuild]: Done in https://github.com/gentoo/gentoo/pull/34844/commits/4823224d61b12fd97a25efcf977885dc04d7904f

File: gui-libs/vte-common/vte-common-0.76.3.ebuild (hunk 6 above)
Comment (flint2) [reply gui-libs/vte-common/vte-common-0.76.3.ebuild]: Done in https://github.com/gentoo/gentoo/pull/34844/commits/01d9d48117482792340739f4f7452598cb90f7f9

File: x11-libs/vte/vte-0.76.3.ebuild (hunk 7 above)
Comment (flint2) [reply x11-libs/vte/vte-0.76.3.ebuild:60]: Done in https://github.com/gentoo/gentoo/pull/34844/commits/3642d1ee1c6b354f4e24275fe3104cfe03acd33a

File: x11-terms/ptyxis/ptyxis-46.3.ebuild (hunk 4 above)
Comment (flint2) [reply x11-terms/ptyxis/ptyxis-46.3.ebuild]: Done in https://github.com/gentoo/gentoo/pull/34844/commits/511e585bf14f96690a668da5a2fd3e26576884e2

**Pull request closed without merging**